### PR TITLE
[FIX] account: fix the default_get of account move lines

### PR DESCRIPTION
**Pull request by len-foss**
Edit journal entries (account.move).
Create a new line. Credit and debit values are filled in with the default get.
Conveniently, if lines_ids is given in context this computes the right value to 
balance existing move lines. However, this is done by float operations. 
Let e be the resulting epsilon in balance.
As a result a new line may contain e, 0 for credit, debit (or respectively).
If the user fills in the 0 value with v, then the new line contains e, v.
This violates the constraint that exactly one of the two values should be nonzero.

We get the currency from the journal to round adequately the balance.

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -489,6 +489,9 @@ def default_get(self, fields):
         for line in self.move_id.resolve_2many_commands(
                 'line_ids', self._context['line_ids'], fields=['credit', 'debit']):
             balance += line.get('debit', 0) - line.get('credit', 0)
+        # if we are here, line_ids is in context, so journal_id should also be.
+        if self._context.get('journal_id') and self.browse(self._context['journal_id']).company_id.currency_id:
```
Comment (nim-odoo) [inline addons/account/models/account_move.py]: I think this is more readable:
```
currency = self._context.get('journal_id') and self.browse(self._context['journal_id']).company_id.currency_id
if currency:
    balance = currency.round(balance)
```

**Comment by len-odoo** [comment]
robodoo rebase-ff r+

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -486,12 +486,13 @@ def default_get(self, fields):
...
-            if line[2]:  # in case of command 0: add a record with values
-                balance += line[2].get('debit', 0) - line[2].get('credit', 0)
-            elif line[0] == 2:  # line has been deleted
-                line_obj = self.browse(line[1])
-                balance -= line_obj.debit - line_obj.credit
+        for line in self.move_id.resolve_2many_commands(
+                'line_ids', self._context['line_ids'], fields=['credit', 'debit']):
+            balance += line.get('debit', 0) - line.get('credit', 0)
+        # if we are here, line_ids is in context, so journal_id should also be.
+        currency = self._context.get('journal_id') and self.browse(self._context['journal_id']).company_id.currency_id
```
Comment (KangOl) [inline addons/account/models/account_move.py]: Looks wrong. `self` is "account.move.line", not "account.journal"
```suggestion
        currency = self._context.get('journal_id') and self.env["account.journal"].browse(self._context['journal_id']).company_id.currency_id
```

This actually break the forward-port to 12.0: http://runbot.odoo.com/runbot/build/596787

Comment (len-odoo) [reply addons/account/models/account_move.py]: It is wrong. https://github.com/odoo/odoo/pull/35939

**Pull request closed without merging**